extension/chrome/settings/setup/setup-key-manager-autogen.ts

```diff
@@ -13,21 +13,33 @@ import { Settings } from '../../../js/common/settings.js';
 import { KeyUtil } from '../../../js/common/core/crypto/key.js';
 import { OpenPGPKey } from '../../../js/common/core/crypto/pgp/openpgp-key.js';
 import { Lang } from '../../../js/common/lang.js';
+import { Xss } from '../../../js/common/platform/xss.js';
 
 export class SetupWithEmailKeyManagerModule {
 
   constructor(private view: SetupView) {
   }
 
   public continueEkmSetupHandler = async () => {
+    const submitButtonSelector = '#step_2_ekm_choose_pass_phrase .action_proceed_private';
+    const submitButton = $(submitButtonSelector);
+    const setBtnColor = (type: 'gray' | 'green') => {
+      submitButton.addClass(type === 'gray' ? 'gray' : 'green');
+      submitButton.removeClass(type === 'gray' ? 'green' : 'gray');
+    };
+    if (! await this.view.isCreatePrivateFormInputCorrect('step_2_ekm_choose_pass_phrase')) {
+      return;
+    }
     try {
-      if (! await this.view.isCreatePrivateFormInputCorrect('step_2_ekm_choose_pass_phrase')) {
-        return;
-      }
+      Xss.sanitizeRender(submitButtonSelector, Ui.spinner('white') + 'Loading...');
+      setBtnColor('gray');
       const passphrase = $('#step_2_ekm_choose_pass_phrase .input_password').val();
       await this.setupWithEkmThenRenderSetupDone(typeof passphrase === 'string' ? passphrase : '');
     } catch (e) {
       await Ui.modal.error(String(e));
+    } finally {
+      setBtnColor('green');
+      submitButton.text('CONTINUE');
```

Comment:
We essentially define the label 'CONTINUE' in 2 places: here and in `.htm` -- not perfect.
It'd be better to make it visible/invisible by class. But we'll need a separate spinner then, similar to `spinner_container` in `add_key.htm`...

Reply:
> Well, i think better to save button text temporary before we set spinner and then recover it after spinner is finished.

I've done that in several places in this codebase - save at the beginning, restore in catch or finally depending on context. It works well and text is defined in a single place.